nextest-runner/src/test_list.rs

```diff
@@ -50,77 +53,93 @@ pub struct RustTestArtifact<'g> {
     pub cwd: Utf8PathBuf,
 }
 
-impl<'g> RustTestArtifact<'g> {
+/// A Rust test binary built by Cargo.
+pub struct RustTestBinary {
+    /// The path to the binary artifact.
+    pub path: Utf8PathBuf,
+    /// The package this artifact belongs to.
+    pub package_id: String,
+    /// The unique binary name defined in `Cargo.toml` or inferred by the filename.
+    pub name: String,
+    /// A unique ID.
+    pub id: String,
+
+    /// Binary build for the host target during cross compilation.
+    /// This include proc-macro tests.
+    pub build_host_only: bool,
```

Comment:
Can this be something like `platform` with an enum with options `host` and `target`?

Reply:
Done